helix-core/src/main/java/org/apache/helix/common/caches/InstanceMessagesCache.java

```diff
@@ -151,6 +153,13 @@ public boolean refresh(HelixDataAccessor accessor, Map<String, LiveInstance> liv
     return true;
   }
 
+  public Map<String, Map<String, Message>> getStaleMessageCache() {
+    return _staleMessageCache;
+  }
+
+  public void setStaleMessageCache(Map<String, Map<String, Message>> staleMessageMap) {
```

Comment:
Not used?

Reply:
If no set is called, how the items in the cache being updated?

Reply:
So it's used by this way:
Map<String, Map<String, Message>> existingStaleMessages = cache.getStaleMessages();
currentStateOutput.setStaleMessageMap(existingStaleMessages);

So when the currentStateOutput's stale message gets updated, the instance cache should also be updated, right? as it's just shallow copy. Let me know if you think differently.

Reply:
This is not a good style. Better to make it explicitly updated.